x-pack/filebeat/input/awss3/metrics.go

```diff
@@ -54,17 +68,46 @@ func (m *inputMetrics) setSQSMessagesWaiting(count int64) {
 	m.sqsMessagesWaiting.Set(count)
 }
 
-func newInputMetrics(id string, optionalParent *monitoring.Registry) *inputMetrics {
+func calculateUtilization(d time.Duration, maxMessagesInflight int, m *inputMetrics) float64 {
```

Comment:
I suggest a naming change to signal that this mutates the data. Like perhaps `calculateUtilizationAndReset`.